numba_cuda/numba/cuda/compiler.py

```diff
@@ -307,8 +307,11 @@ def compile(pyfunc, sig, debug=False, lineinfo=False, device=True,
     if output not in ("ptx", "ltoir"):
         raise NotImplementedError(f'Unsupported output type: {output}')
 
+    debug = config.CUDA_DEBUGINFO_DEFAULT if debug is None else debug
```

Comment:
Whilst I think this change makes sense, I think `debug=False` needs changing to `debug=None` in the kwargs as well if we're making this change.

Reply:
(I think a similar change would need making for the other `compile_` functions beneath, too)

Reply:
change `debug=None` in signature for `compile` and 3 wrappers.